pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/util/DSMOncHistoryCreatorUtil.java

```diff
@@ -0,0 +1,258 @@
+package org.broadinstitute.dsm.util;
+
+
+import static org.broadinstitute.dsm.service.admin.UserAdminService.USER_ADMIN_ROLE;
+import static org.broadinstitute.dsm.statics.DBConstants.MR_VIEW;
+import static org.mockito.Mockito.mock;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import net.minidev.json.JSONObject;
+import org.broadinstitute.dsm.db.OncHistoryDetail;
+import org.broadinstitute.dsm.db.Tissue;
+import org.broadinstitute.dsm.db.dao.DeletedObjectDao;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dao.ddp.institution.DDPInstitutionDao;
+import org.broadinstitute.dsm.db.dao.ddp.medical.records.MedicalRecordDao;
+import org.broadinstitute.dsm.db.dao.ddp.participant.ParticipantDao;
+import org.broadinstitute.dsm.db.dao.ddp.tissue.TissueDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.db.dto.ddp.participant.ParticipantDto;
+import org.broadinstitute.dsm.model.patch.BasePatch;
+import org.broadinstitute.dsm.model.patch.Patch;
+import org.broadinstitute.dsm.model.patch.PatchFactory;
+import org.broadinstitute.dsm.service.admin.UserAdminTestUtil;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.junit.Assert;
+import org.mockito.Mock;
+import org.testcontainers.shaded.org.apache.commons.lang3.StringUtils;
+
+/**
+ * This class is to create participants with Medical Record and Onc Histories in DSM
+ * It utilises the UserAdminTestUtil to create a ddpInstance and a user with privileges
+ * and to use that to be
+ **/
+@Slf4j
+public class DSMOncHistoryCreatorUtil {
```

Comment:
I think the naming patter we are using is *TestUtil, so this should be `DSMOncHIstoryCreatorUtil` or `DSMOncHistoryTestUtil`.